### PR TITLE
switch travis to use autograph docker image

### PR DESCRIPTION
**Pull request by g-k**
PR to fix https://github.com/mozilla/addons-server/issues/9008

Changes:

* replace scripts/start-autograph.sh (using go get, gimme) with the autograph docker image from https://hub.docker.com/r/mozilla/autograph/tags/

Why not fix go get?

* I'm still not sure why it's failing.
* Autograph 2.2.0 depends on the ltdl C library to talk to HSMs and we don't want to have to install that and other dependencies in the future https://github.com/mozilla-services/autograph/blob/master/docs/hsm.rst
* We can cache the built docker image to speed up builds

### CODE REVIEW

**Comment by EnTeQuAk** [comment]
> We can cache the built docker image to speed up builds

For a small image like this, this probably won't change anything. The cache is restored via network anyway which isn't much faster compared to simply downloading the docker image.

File: .travis.yml [hunk 1]
```diff
@@ -65,7 +65,7 @@ script:
   - |
     if [[ $TRAVIS_EVENT_TYPE != "cron" ]]; then
       if [[ $TOXENV == "amo-locales-and-signing" ]]; then
-         ./scripts/start-autograph.sh
+         docker run --name autograph -d -p 5500:5500 -v $(pwd)/scripts/:/scripts/ mozilla/autograph:2.0.5 /go/bin/autograph -c /scripts/autograph_travis_test_config.yaml
```
Comment (EnTeQuAk) [inline .travis.yml]: This should be hitting `autograph:latest`, most certainly not `2.0.5` when `2.2.0` is current.

**Comment by g-k** [comment]
The test failures look unrelated to autograph, so I think this is safe to land.

**Comment by EnTeQuAk** [comment]
> The test failures look unrelated to autograph, so I think this is safe to land.

I restarted the devhub task, let's see. We usually don't run PRs with failing tests except for CircleCI which is optionally at times.

**Pull request merged**